code/modules/recycling/conveyor.dm

```diff
@@ -390,10 +400,18 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	update_linked_switches()
 
 
-/obj/machinery/conveyor_switch/attackby(obj/item/attacking_item, mob/user, params)
-	if(is_wire_tool(attacking_item))
-		wires.interact(user)
-		return TRUE
+/obj/machinery/conveyor_switch/wirecutter_act(mob/living/user, obj/item/I)
+	wires.interact(user)
+	return TRUE
```

Comment:
it seems as if conveyor belts have logic that's primarily for pulses. in this case it's for assemblies I suppose.

On an unrelated note, it would be cool if you could keep the conveyor switch wire datum from holding a reference to the last mob that interacted with it. Don't need to, but I'd really appreciate it. Would want to use a weakref

Reply:
This should also not be on wirecutter act. More things then wirecutters can interact with wires

Reply:
wait so what do you want me to do with the logic and pulses? just curious

Reply:
also Im not sure how to use WEAKREFs

Reply:
See discord.
Weakrefs have a pretty good description of what they do/how to use them in their documentation, is it tripping you up?